File: do_not_commit/useless-ifelse.py
a, b, c = 1

# ruleid: useless-if-conditional
if a:
    print('1')
elif a:
    print('2')

# ruleid: useless-if-body
if a:
    print('1')
else:
    print('1')

# a and b are different cases -- ok
if a:
    print('1')
elif b:
    print('1')


# don't report on cases like this
if a:
    print('this is a')
elif b:
    print('this is b')
elif c:
    print('this is c')
elif d:
    print('this is d')


# don't report on cases like this
if a:
    print('this is a')
elif b:
    print('this is b')
elif c:
    print('this is b')

File: do-not-commit/useless-ifelse.go
package main

import "fmt"

func main() {
	fmt.Println("hello world")
	var y = 1

	if y {
		fmt.Println("of course")
	}

	// ruleid:useless-if-conditional
	if y {
		fmt.Println("of course")
	} else if y {
		fmt.Println("of course other thing")
	}

	// ruleid:useless-if-body
	if y {
		fmt.Println("of course")
	} else {
		fmt.Println("of course")
	}

	fmt.Println("of course2")
	fmt.Println(1)
	fmt.Println(2)
	fmt.Println(3)
	fmt.Println("of course2")

}
